components/common/folderPage/modal/Modal.tsx

```diff
@@ -1,143 +1,77 @@
-  share?: boolean;
-}) {
-  const { setClickedOption } = useContext(FolderContext);
-
-  return (
-    <>
-      <Background>
-        <Container>
-          <TitleContainer>
-            <Title>{title}</Title>
-            {subtitle && <SubTitle>{subtitle}</SubTitle>}
-          </TitleContainer>
-          <ButtonContainer>
-            <Image
-              src={Close}
-              width={24}
-              height={24}
-              alt="close-icons"
-              onClick={() => {
-                setClickedOption({
-                  addLink: false,
-                  shareLink: false,
-                  editLinkName: false,
-                  deleteLink: false,
-                });
-              }}
-            />
-          </ButtonContainer>
-          <InputContainer>
-            {input && <ModalInput currentFolder={folderName} />}
-            {button?.color === 'red' && (
-              <ModalButton color={button.color} text={button.text} />
-            )}
-          </InputContainer>
-          {folderLists && <FolderList folderLists={folderLists} />}
-          {button?.color === 'blue' && (
-            <ModalButton color={button.color} text={button.text} />
-          )}
-          {share && <IconsBox />}
-        </Container>
-      </Background>
-    </>
-  );
-}
+import { useContext } from 'react';
+import ModalInput from './ModalInput';
+import ModalButton from './ModalButton';
+import IconsBox from './IconsBox';
+import FolderList from './FolderList';
+import FolderContext from '../../../../contexts/FolderContext';
+import Image from 'next/image';
+import Close from '/public/assets/icons/modal/close.svg';
+import styles from '@/styles/folderPage/modal/modal.module.css';
+
+export default function Modal({
+  title,
+  input = false,
+  button,
+  folderName,
+  subtitle = '',
+  folderLists,
+  share,
+  closeKebab,
+}: {
+  title: string;
+  input?: boolean;
+  button?: {
+    color: string;
+    text: string;
+  };
+  folderName?: string;
+  subtitle?: string;
+  folderLists?: object[];
+  share?: boolean;
+  closeKebab?: any;
+}) {
+  const { setClickedOption } = useContext(FolderContext);
+
+  return (
+    <>
+      <div className={styles.background}>
+        <div className={styles.container}>
+          <div className={styles.titleWrapper}>
+            <span className={styles.title}>{title}</span>
+            {subtitle && <span className={styles.subtitle}>{subtitle}</span>}
+          </div>
+          <button className={styles.button}>
+            <Image
+              src={Close}
+              width={24}
+              height={24}
+              alt="close-icons"
+              onClick={() => {
+                setClickedOption({
+                  addLink: false,
+                  shareLink: false,
+                  editLinkName: false,
+                  deleteLink: false,
+                });
+                if (closeKebab) {
+                  closeKebab();
+                }
```

Comment:
큰 차이는 없지만 if 구문 없이 옵셔널 체이닝을 함수 호출에도 적용할 수 있습니다.
```suggestion
                closeKebab?.();
```

Reply:
오! 옵셔널 체이닝을 이렇게도 쓸 수 있군요! 감사합니다.